src/commands/skip.rs

```diff
@@ -20,9 +23,43 @@ pub async fn skip(
 
     if queue.is_empty() {
         return create_response(&ctx.http, interaction, NOTHING_IS_PLAYING).await;
-    } else if queue.skip().is_ok() {
-        return create_response(&ctx.http, interaction, SKIPPED).await;
     }
 
+    let mut data = ctx.data.write().await;
+    let cache_map = data.get_mut::<GuildCacheMap>().unwrap();
+
+    let cache = cache_map.entry(guild_id).or_default();
+
+    if !cache.current_skip_votes.contains(&interaction.user.id) {
+        cache.current_skip_votes.push(interaction.user.id);
+    }
+
+    let guild = ctx.cache.guild(guild_id).await.unwrap();
+    let skip_threshold = guild.voice_states.len() / 2;
+
+    let message = if cache.current_skip_votes.len() >= skip_threshold && queue.skip().is_ok() {
```

Comment:
I'd rather have `queue.skip()` checked for OK inside the condition, it's not very readable in the current state.
Additionally, if skipping fails, it displays the skip vote count which I'm not certain it is what you want.

Reply:
In that case I won't be able to remove the last Ok(()), since I won't always have a `message`.

Reply:
That's alright!